lib/ansible/vars/manager.py

```diff
@@ -487,7 +487,7 @@ def _get_magic_variables(self, play, host, task, include_hostvars, include_deleg
             variables['groups'] = self._inventory.get_groups_dict()
             if play:
                 templar = Templar(loader=self._loader)
-                if templar.is_template(play.hosts):
+                if not play.finalized and templar.is_template(play.hosts):
```

Comment:
I think this roughly needs replicated into `lib/ansible/plugins/strategy/__init__.py` also.